icaruscode/IcarusObj/PMTBeamSignal.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file   icaruscode/IcarusObj/PMTBeamSignal.h
+ * @brief  Holds the event-by-event RWM or EW times
+ * @author Matteo Vicenzi (mvicenzi@bnl.gov)
+ * @date   March 14 2024
+ */
+
+#ifndef ICARUSCODE_ICARUSOBJ_PMTBEAMSIGNAL_H
+#define ICARUSCODE_ICARUSOBJ_PMTBEAMSIGNAL_H
+
+// C/C++ standard libraries
+#include <limits>
+
+namespace icarus::timing{
+		
+        /// Special value to denote no special channel information
+        static constexpr auto NoChannel = std::numeric_limits<unsigned int>::max();
+        /// Special value to denote no time channel information
+        static constexpr double NoTime = 0.;
+        /// Special value to denote no sample information
+        static constexpr size_t NoSample = 0;
+
+	struct PMTBeamSignal {
```

Comment:
Add a brief description of the content and purpose of the data product.
For example:
```cpp
    /**
     * @brief Beam extraction time as seen by a PMT readout board.
     * 
     * Includes both time in @ref DetectorClocksElectronicsTime "electronics time scale"
     * and time relative to the hardware trigger.
     * 
     * The information in this object may be missing: its validity should
     * always be checked in advance with `isValid()`.
     */
```
Here I integrate also a comment from later about the validity of the data.